cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -133,17 +134,40 @@ func NewMetricsBuilder(settings MetricsSettings, options ...metricBuilderOption)
 // Emit appends generated metrics to a pdata.MetricsSlice and updates the internal state to be ready for recording
 // another set of data points. This function will be doing all transformations required to produce metric representation
 // defined in metadata and user settings, e.g. delta/cumulative translation.
-func (mb *MetricsBuilder) Emit(metrics pdata.MetricSlice) {
+func (mb *MetricsBuilder) Emit() pdata.Metrics{
+	if mb.data == nil {
+		return pdata.NewMetrics()
+	}
 	{{- range $name, $metric := .Metrics }}
-	mb.metric{{- $name.Render }}.emit(metrics)
+	mb.metric{{- $name.Render }}.emit(mb.data.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics())
 	{{- end }}
+	return *mb.data
+}
+
+// EnsureCapacity TODO
+func (mb *MetricsBuilder) EnsureCapacity(length int) {
```

Comment:
I don't think we need these functions. Capacity can be always taken from number of enabled metrics

Reply:
Or we can follow similar approach to how we figure out capacity for datapoints from the first scrape.